File: 005-reverse-vowel-of-a-string/Solution.java
import java.util.*;

class Solution {
    public String reverseVowels(String s) {
        Set<Character> vowels = new HashSet<>(Arrays.asList('a', 'e', 'i', 'o', 'u'));
        char[] characters = s.toCharArray();

        int left = 0;
        int right = characters.length - 1;

        while(left < right) {
            while(left < right && !vowels.contains(Character.toLowerCase(characters[left]))) {
                left += 1;
            }

            while(left < right && !vowels.contains(Character.toLowerCase(characters[right]))) {
                right -= 1;
            }

            swapAt(left, right, characters);

            left += 1;
            right -= 1;
        }

        return new String(characters);
    }

    private void swapAt(int i, int j, char[] characters) {
        char temp = characters[i];
        characters[i] = characters[j];
        characters[j] = temp;
    }
}

File: 005-reverse-vowel-of-a-string/Solution.swift
class Solution {
    func reverseVowels(_ s: String) -> String {
        var vowels: Set<Character> = ["a", "e", "i", "o", "u"]
        var characters: [Character] = Array(s)
        var left = 0
        var right = characters.count - 1

        while left < right {
            while left < right, !vowels.contains { String($0) == characters[left].lowercased() } {
                left += 1
            }

            while left < right, !vowels.contains { String($0) == characters[right].lowercased() } {
                right -= 1
            }

            characters.swapAt(left, right)
            left += 1
            right -= 1
        }
        return String(characters)
    }
}